### PR TITLE
Fixed #35809 -- Set background color for selected rows in the admin's form select widget.

### PR DESCRIPTION
**Pull request by nessita**
Regression in b47bdb4cd9149ee2a39bf1cc9996a36a940bd7d9.

Thank you Giannis Terzopoulos for the review.

#### Trac ticket number

ticket-35809

#### Branch description
This branch provides explicitly in the CSS definitions the color for the form's select selected row. See before (in both screenshots there are selected rows in the multi-select with focus somewhere else):
image
image

And after:
image
image

#### Checklist
- [X] This PR targets the `main` branch. 
- [X] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [X] I have checked the "Has patch" ticket flag in the Trac system.
- [ ] I have added or updated relevant tests.
- [X] I have added or updated relevant docs, including release notes if applicable.
- [X] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

**Review by knyghty — APPROVED** [review]
It looks good. I am not completely sure if it would affect anything anywhere else. I suspect not in an obviously bad way, at least.

There is a space missing in the PR title.

**Comment by nessita** [comment]
> It looks good. I am not completely sure if it would affect anything anywhere else. I suspect not in an obviously bad way, at least.

Thank you!

> There is a space missing in the PR title.

:see_no_evil:  fixed :-)

**Review by sarahboyce — COMMENTED** [review]
```diff
--- a/tests/admin_views/tests.py
+++ b/tests/admin_views/tests.py
@@ -6125,6 +6125,7 @@ class SeleniumTests(AdminSeleniumTestCase):
             password="secret",
             login_url=reverse("admin7:index"),
         )
+        cheddar = Topping.objects.create(name="Cheddar")
         url = self.live_server_url + reverse("admin7:admin_views_pizza_add")
         self.selenium.get(url)
         self.selenium.find_elements(By.TAG_NAME, "summary")[0].click()
@@ -6142,7 +6143,12 @@ class SeleniumTests(AdminSeleniumTestCase):
                 + from_box.get_property("offsetHeight")
             ),
         )
-        self.take_screenshot("selectbox-collapsible")
+        self.take_screenshot("selectbox-non-selected")
+        cheddar_option = self.selenium.find_element(
+            By.CSS_SELECTOR, f"option[value='{cheddar.id}']"
+        )
+        cheddar_option.click()
+        self.take_screenshot("selectbox-selected")

     @screenshot_cases(["desktop_size", "mobile_size", "rtl", "dark", "high_contrast"])
```
Be nice to add a screenshot to our test suite 👍

File: docs/releases/5.1.2.txt [hunk 1]
```diff
@@ -15,3 +15,6 @@ Bugfixes
 
 * Fixed a regression in Django 5.1 that caused a crash of ``JSONObject()``
   when using server-side binding with PostgreSQL 16+ (:ticket:`35734`).
+
+* Fixed a regression in Django 5.1 that caused an accessibility issue with the
+  admin's form select widget in dark theme (:ticket:`35809`).
```
[the remark this answers is not in the document]
Comment (nessita) [reply docs/releases/5.1.2.txt:20]: Thanks, applied a small variation of this!

**Comment by nessita** [comment]
> ```diff
> --- a/tests/admin_views/tests.py
> +++ b/tests/admin_views/tests.py
> @@ -6125,6 +6125,7 @@ class SeleniumTests(AdminSeleniumTestCase):
>              password="secret",
>              login_url=reverse("admin7:index"),
>          )
> +        cheddar = Topping.objects.create(name="Cheddar")
>          url = self.live_server_url + reverse("admin7:admin_views_pizza_add")
>          self.selenium.get(url)
>          self.selenium.find_elements(By.TAG_NAME, "summary")[0].click()
> @@ -6142,7 +6143,12 @@ class SeleniumTests(AdminSeleniumTestCase):
>                  + from_box.get_property("offsetHeight")
>              ),
>          )
> -        self.take_screenshot("selectbox-collapsible")
> +        self.take_screenshot("selectbox-non-selected")
> +        cheddar_option = self.selenium.find_element(
> +            By.CSS_SELECTOR, f"option[value='{cheddar.id}']"
> +        )
> +        cheddar_option.click()
> +        self.take_screenshot("selectbox-selected")
> 
>      @screenshot_cases(["desktop_size", "mobile_size", "rtl", "dark", "high_contrast"])
> ```
> 
> Be nice to add a screenshot to our test suite 👍

Good idea!

I have added a separated test exercising the flow of selecting multiple items from available option list, and also from the "chosen" option list, also because the bug was more evident when the focus was "moved" to another element without really actioning on them.

The separation is mostly to avoid polluting the tests about fieldsets (collapsible/non collapsible).

File: tests/admin_views/tests.py [hunk 2]
```diff
@@ -6175,6 +6175,59 @@ def test_selectbox_height_not_collapsible_fieldset(self):
...
+        perms = list(Permission.objects.filter(content_type=ct))
+        for perm in perms:
+            elem = self.selenium.find_element(
+                By.CSS_SELECTOR, f"#id_user_permissions_from option[value='{perm.id}']"
+            )
+            ActionChains(self.selenium).key_down(Keys.CONTROL).click(elem).key_up(
+                Keys.CONTROL
+            ).perform()
+
+        ActionChains(self.selenium).send_keys(Keys.TAB).perform()  # Move focus
```
Comment (sarahboyce) [inline tests/admin_views/tests.py:6210]: test_selectbox_selected_rows--selectbox-chosen-perms-none-selected--desktop_size

Can we tweak this to click on or focus on something else (maybe something explicit, like the box)? The button has an animation and so it looks like we have an issue with the icon (but it just catches it while it's moving)

Comment (nessita) [reply tests/admin_views/tests.py]: Sure, I just push a new revision where the focus is moved to the filter input. Screenshots:
test_selectbox_selected_rows--selectbox-available-perms-some-selected--dark

File: tests/admin_views/tests.py [hunk 3]
```diff
@@ -6175,6 +6175,65 @@ def test_selectbox_height_not_collapsible_fieldset(self):
...
+            )
+            ActionChains(self.selenium).key_down(Keys.CONTROL).click(elem).key_up(
+                Keys.CONTROL
+            ).perform()
+
+        # Move focus to other element.
+        elem = self.selenium.find_element(
+            By.CSS_SELECTOR, "#id_user_permissions_selected_input"
+        )
+        ActionChains(self.selenium).move_to_element(elem).click().perform()
```
Comment (sarahboyce) [inline tests/admin_views/tests.py:6234]:
```suggestion
        self.selenium.find_element(
            By.CSS_SELECTOR, "#id_user_permissions_selected_input"
        ).click()
```

File: tests/admin_views/tests.py [hunk 4]
```diff
@@ -6175,6 +6175,65 @@ def test_selectbox_height_not_collapsible_fieldset(self):
...
+            elem = self.selenium.find_element(
+                By.CSS_SELECTOR, f"#id_user_permissions_from option[value='{perm.id}']"
+            )
+            ActionChains(self.selenium).key_down(Keys.CONTROL).click(elem).key_up(
+                Keys.CONTROL
+            ).perform()
+
+        # Move focus to other element.
+        elem = self.selenium.find_element(By.CSS_SELECTOR, "#id_user_permissions_input")
+        ActionChains(self.selenium).move_to_element(elem).click().perform()
```
Comment (sarahboyce) [inline tests/admin_views/tests.py:6212]:
```suggestion
        self.selenium.find_element(
            By.CSS_SELECTOR, "#id_user_permissions_input"
        ).click()
```
(nit) we don't need to do an action chain here

**Review by sarahboyce — APPROVED** [review]
Thank you!

File: tests/admin_views/tests.py (hunk 4 above)
Comment (nessita) [reply tests/admin_views/tests.py:6212]: You are right, removing.

**Pull request merged**